components/merkledb/src/proof_list_index/proof.rs

```diff
@@ -309,3 +330,69 @@ where
         Ok(res)
     }
 }
+
+impl<V: Serialize> Serialize for ListProof<V> {
+    fn serialize<S>(&self, ser: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = ser.serialize_struct("Proof", 2)?;
+        state.serialize_field("length", &self.length)?;
+        state.serialize_field("proof", &self.proof)?;
+        state.end()
+    }
+}
+impl<'a, V> Deserialize<'a> for ListProof<V>
+where
+    V: BinaryValue,
+    for<'de> V: Deserialize<'de>,
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'a>,
+    {
+        fn format_err_string(type_str: &str, value: &Value, err: &SerdeJsonError) -> String {
+            format!(
+                "Couldn't deserialize {} from serde_json::Value: {}, error: {}",
+                type_str, value, err
+            )
+        }
+
+        let json: Value = <Value as Deserialize>::deserialize(deserializer)?;
```

Comment:
This code looks very strange for me. We use generic serialization and concrete json for deserialization. 
It seems that we should write a special serde wrapper structure for this one.

Can you add `TODO` with the `[ECR-3222]`  marker about this problem or fix it if fix is trivial?